src/OrchardCore.Modules/OrchardCore.Layers/Migrations.cs

```diff
@@ -29,5 +56,29 @@ public int UpdateFrom1()
 
             return 2;
         }
+
+        public async Task<int> UpdateFrom2Async()
+        {
+            var layers = await _layerService.LoadLayersAsync();
+            foreach (var layer in layers.Layers)
+            {
+                layer.LayerRule = new Rule();
+                _conditionIdGenerator.GenerateUniqueId(layer.LayerRule);
+
+                #pragma warning disable 0618
+                _ruleMigrator.Migrate(layer.Rule, layer.LayerRule);
+
+                layer.Rule = String.Empty;
+                #pragma warning restore 0618
+            }
+
+            await _layerService.UpdateAsync(layers);
+
+            var feature = _extensionManager.GetFeatures().FirstOrDefault(f => f.Id == "OrchardCore.Rules");
+            
+            await _shellFeaturesManager.EnableFeaturesAsync(new[] { feature }, force: true);
```

Comment:
@jtkech your thoughts here if you have some time?

Despite taking a dependency in the `Manifest` for old installations, I have still had to activate the new `OrchardCore.Rules` feature manually here.

That surprised me?

And this may not be the best way to force activation of the new feature

Reply:
@deanmarcussen Yes, a dependency is automatically enabled when you enable the dependent through a recipe or through the admin, but not when an already enable feature has a new dependency. Not usual that a module upgrade adds a dependency, we already did that but for missing deps related to features commonly enabled through a recipe.

> And this may not be the best way to force activation of the new feature

Looks good, as i remember here you could just enable `OC.Layers`  that woud enable its dependencies and then the missing one. Hmm, without magic string you could also use `ITypeFeatureProvider` to get the layer feature.

    var layerFeature = _typeFeatureProvider.GetFeatureForDependency(migration.GetType());
    _shellFeaturesManager.EnableFeaturesAsync(new[] { layerFeature }, force: true);

Reply:
Done. nice cheers - works better. I like that